services/workflows-service/src/common/get-file-extension-with-fallback/get-file-extension-with-fallback.ts

```diff
@@ -0,0 +1,38 @@
+import { getFileExtension } from '@/common/get-file-extension/get-file-extension';
+import mime from 'mime';
+import { getMimeType } from '@/common/get-mime-type/get-mime-type';
+import { getFileType } from '@/common/get-file-type/get-file-type';
+
+/**
+ * Returns file's extension and mime type with fallback for each.
+ * @param file - file's path or buffer
+ * @param fileName - file's name with extension
+ * @param preferFileName - if false, will prefer extracting the file's extension from the file's mime type over file's name
+ */
+export const getFileExtensionWithFallback = async ({
+  file,
+  fileName,
+  preferFileName = true,
```

Comment:
in case someone call you function like that:
getFileExtensionWithFallback({ file: 'aa', fileName: 'bb'}) your preferFileName will be present with default value?

Reply:
{ ...DEFAULT_OPTIONS, ...options }

Reply:
Default values work as intended.